web/public/js/controllers/manage-dojo-users-controller.js

```diff
@@ -60,11 +62,237 @@ function cdManageDojoUsersCtrl($scope, $state, $q, cdDojoService, alertService,
+        if (user) {
+          return user.id !== $scope.currentUser.id;
+        } else {
+          return false;
+        }
+      }
+    },
+    viewProfile: {
+      ngShow: function () {
+        return $scope.selectedItems.length === 1;
+      },
+      ngHref: function () {
+        var users = $scope.selectedItems;
+        if (users.length === 1) {
+          return '/profile/' + users[0].userData.id;
+        } else {
+          return null;
+        }
+      }
+    },
+    changeRole: {
+      ngShow: function () {
+        return $scope.selectedItems.length === 1;
+      },
+      ngClick: function () {
+        var user = $scope.selectedItems[0].userData;
+        var age = moment.utc().diff(moment(user.dob), 'years');
+        var roles = $scope.adultRoles;
+        if (age < 18) {
+          roles = $scope.youthRoles;
+        }
+        cdChooseRoleModal({
+          roles: roles,
+          callback: function (role) {
+            return $q(function (resolve, reject) {
+              var hasPermission = false;
+              var userType = role.userType;
+              canUpdateUser(function (result) {
+                hasPermission = result;
+
+                if(hasPermission) {
+
+                  var userDojoLink = _.find(usersDojosLink, {userId:user.id});
+                  if(!userDojoLink.userTypes) userDojoLink.userTypes = [];
+                  var roles = [];
+                  if (age < 13) {
+                    roles.push('attendee-u13');
+                  } else if (age < 18) {
+                    roles.push('attendee-o13');
+                  }
+                  if (userType && userType.name) {
+                    roles.push(userType.name);
+                  }
+                  $scope.selectedItems[0].subCaption = userUtils.getTitleForUserTypes(roles, user);
+                  userDojoLink.userTypes = roles;
+                  cdDojoService.saveUsersDojos(userDojoLink, function (response) {
+                    if(response.error) {
+                      reject($translate.instant(response.error));
+                    } else {
+                      resolve($translate.instant('User types successfully updated.'));
+                    }
+                  }, function (err) {
+                    reject($translate.instant('Error saving user type') + ' ' + err);
+                  });
+                } else {
+                  reject($translate.instant('You do not have permission to update user types'));
+                }
+              });
+            });
+          },
+          title: $translate.instant('Change Role')
+        });
+      }
+    },
+    remove: {
+      ngClick: function () {
+        var user = $scope.selectedItems[0].userData;
+        $ngBootbox.confirm($translate.instant('Are you sure you want to remove this user from your Dojo?'))
+          .then(function() {
+            $scope.removeUser(user);
```

Comment:
removeUser does not clearSelection, as clearSelection is only invoked when you select another user, am i right ?